File: src/ios/FamilyHub/Networking/CalendarAPI.swift
import Foundation

extension APIClient {
    func getMyEvents() async throws -> [CalendarEvent] {
        try await get("api/events")
    }

    func getFamilyEvents(familyId: UUID) async throws -> [FamilyCalendarEvent] {
        try await get("api/families/\(familyId)/events")
    }

    func createEvent(_ request: CreateCalendarEventRequest) async throws -> CalendarEvent {
        try await post("api/events", body: request)
    }

    func deleteEvent(id: UUID) async throws {
        try await deleteResource("api/events/\(id)")
    }
}

File: src/frontend/src/app/api/calendarApi.js
import { apiClient } from './client';
export async function getMyEvents() {
    const response = await apiClient.get('/events');
    return response.data;
}
export async function createEvent(request) {
    const response = await apiClient.post('/events', request);
    return response.data;
}
export async function getFamilyEvents(familyId) {
    const response = await apiClient.get(`/families/${familyId}/events`);
    return response.data;
}
